x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/support/MetadataUtils.java

```diff
@@ -10,11 +10,21 @@
 
 public class MetadataUtils {
 
+    public static final String RESERVED_ROLE_VERSION_METADATA_KEY = "reserved_role_version";
     public static final String RESERVED_PREFIX = "_";
     public static final String RESERVED_METADATA_KEY = RESERVED_PREFIX + "reserved";
     public static final String DEPRECATED_METADATA_KEY = RESERVED_PREFIX + "deprecated";
     public static final String DEPRECATED_REASON_METADATA_KEY = RESERVED_PREFIX + "deprecated_reason";
-    public static final Map<String, Object> DEFAULT_RESERVED_METADATA = Map.of(RESERVED_METADATA_KEY, true);
+    /* !!!!!! IMPORTANT !!!!!!!
+     * reserved role version must be incremented when builtin reserved roles are changed
+     */
+    public static final String RESERVED_ROLE_VERSION_VALUE = "2";
```

Comment:
We need the version so that we index, but not overwrite, built-in reserved roles after they're changed.